auth/service.cc

```diff
@@ -418,6 +426,232 @@ future<bool> service::exists(const resource& r) const {
     return make_ready_future<bool>(false);
 }
 
+future<std::vector<cql3::description>> service::describe_roles(bool with_salted_hashes) const {
+    std::vector<cql3::description> result{};
+
+    const auto roles = co_await _role_manager->query_all();
+    result.reserve(roles.size());
+
+    const bool authenticator_uses_salted_hashes = _authenticator->uses_salted_hashes();
+
+    auto maybe_fetch_salted_hash = [&] (std::string_view role_name) -> future<std::optional<sstring>> {
```

Comment:
why this lambda is needed? wouldn't be simpler to write later, when you iterate over roles:

std::optional<sstring> salted_hash;
if (authenticator_uses_salted_hashes) {
salted_hash = ...query_salted_hash(...)
}

?